share/spice/snow/snow.js

```diff
@@ -53,202 +53,104 @@ function nrio (api_result) {
-                break;
-            }
-        }
-    
-        var x = (-300 + Math.floor(Math.random() * (ww + 300)));
-        var y = 0;
+    requestAnimFrame(update);
+  }
 
-        if (typeof initial == 'undefined' || !initial) {
-            y = -300;
-        } else {
-            y = (-300 + Math.floor(Math.random() * (wh + 300)));
-        }
-    
-        $nowflake.css({
-            left: x + 'px',
-            top: y + 'px'
-        });
-        
-        $nowflake.data('x', x);
-        $nowflake.data('y', y);
-        $nowflake.data('v', v);
-        $nowflake.data('half_v', Math.round(v * 0.5));
-        
-        $dom.append($nowflake);
-    }
-    
-    function move() {
-        if (Math.random() > 0.8) {
-            xv += -1 + Math.random() * 2;
-            
-            if (Math.abs(xv) > 3) {
-                xv = 3 * (xv / Math.abs(xv));
-            }
-        }
-        
-        // Throttle code
-        var newTime = new Date().getTime();
-        var diffTime = newTime - prevTime;
-        prevTime = newTime;
-        
-        if (diffTime < 55 && flakeCount < absMax) {
-            addFlake();
-        } else if (diffTime > 150) {
-            $('span.winternetz:first').remove();
-            flakeCount--;
-        }
-        
-        $('span.winternetz').each(function () {
-            var x = $(this).data('x');
-            var y = $(this).data('y');
-            var v = $(this).data('v');
-            var half_v = $(this).data('half_v');
-            
-            y += v;
-            
-            x += Math.round(xv * v);
-            x += -half_v + Math.round(Math.random() * v);
-            
-            // because flakes are rotating, the origin could be +/- the size of the flake offset
-            if (x > maxw) {
-                x = -300;
-            } else if (x < minw) {
-                x = ww;
-            }
-            
-            if (y > maxh) {
-                $(this).remove();
-                flakeCount--;
-                addFlake();
-            } else {
-                $(this).data('x', x);
-                $(this).data('y', y);
-
-                $(this).css({
-                    left: x + 'px',
-                    top: y + 'px'
-                });
-                
-                // only spin biggest three flake sizes
-                if (v >= 6) {
-                    $(this).animate({rotate: '+=' + half_v + 'deg'}, 0);
-                }
-            }
-            }
-        );
-    }
+  // shim layer with setTimeout fallback
+  window.requestAnimFrame = (function(){
+    return  window.requestAnimationFrame       ||
+            window.webkitRequestAnimationFrame ||
+            window.mozRequestAnimationFrame    ||
+            function( callback ){
+              window.setTimeout(callback, 1000 / 60);
+            };
+  })();
 
+  onResize();
+  window.addEventListener('resize', onResize, false);
 
-    makeItSnow();
+  masthead.appendChild(canvas);
+}
+ makeItSnow();
```

Comment:
Also, you don't want to fire the event here. You'll want to use the [onShow event](https://duck.co/duckduckhack/display_reference#Events).